### PR TITLE
Tolerate nil input in GetValueFromIntOrPercent

### PR DESCRIPTION
**Pull request by soltysh**
**What this PR does / why we need it**:
`GetValueFromIntOrPercent` accepts pointer argument but does not validate it. This PR fixes that problem preventing from panics.

/assign @deads2k @sttts 

**Release note**:
```release-note
NONE
```

### CODE REVIEW

File: staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go [hunk 1]
```diff
@@ -143,6 +143,9 @@ func (intstr *IntOrString) Fuzz(c fuzz.Continue) {
 }
 
 func GetValueFromIntOrPercent(intOrPercent *IntOrString, total int, roundUp bool) (int, error) {
+	if intOrPercent == nil {
+		return 0, nil
```
Comment (sttts) [inline staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go]: What was the old behaviour? panic? error?

This func needs godoc.

Comment (soltysh) [reply staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go]: Panic, although when dealing with internal APIs you didn't have that problem b/c usually these fields are non-pointer. Now that we're slowly going towards external APIs I passed in pointer and it failed terribly.

Comment (deads2k) [reply staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go]: I don't care for this kind of value compression.  Doing this makes nil indistinguishable from `0`.  I'd rather return an error and the places you need to tolerate the nil, you can have a `intOrPercentPtrOrDefault(zero)` that you pass for the first value if that's the behavior you need/want

**Comment by soltysh** [comment]
@deads2k update, ptal
@janetkuo @tnozicka  fyi the second commit

File: staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go [hunk 2]
```diff
@@ -143,6 +144,9 @@ func (intstr *IntOrString) Fuzz(c fuzz.Continue) {
 }
```
Comment (deads2k) [inline staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go]: If you had a 

```go
func NewZero() *IntOrPercent{
    return the equivalent of zero
}

ValOrDefault(intOrPercent *IntOrString, defaultValue *IntOrString){
  if intOrPercent == nil{
       return defaultValue
    }
}
```

Then you could use it in the above case and whereever it is in downstream too, right?

Comment (soltysh) [reply staging/src/k8s.io/apimachinery/pkg/util/intstr/intstr.go]: Yeah, that make sense. Althogh `NewZero` is not needed we have `FromInt` and `FromString` helpers that should do it.

**Pull request merged**